internal/services/apimanagement/api_management_product_tag_resource_test.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ApiManagementProductTagResource struct{}
+
+func TestAccApiManagementProductTag_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_product_tag", "test")
+	r := ApiManagementProductTagResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("product_name").Exists(),
+				check.That(data.ResourceName).Key("api_management_name").Exists(),
+				check.That(data.ResourceName).Key("resource_group_name").Exists(),
+				check.That(data.ResourceName).Key("tag_name").Exists(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiManagementProductTag_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_product_tag", "test")
+	r := ApiManagementProductTagResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (ApiManagementProductTagResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.ProductTagID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ApiManagement.TagClient.GetByProduct(ctx, id.ResourceGroup, id.ServiceName, id.ProductName, id.TagName)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.ID != nil), nil
+}
+
+func (ApiManagementProductTagResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+
+  sku_name = "Consumption_0"
+}
+
+resource "azurerm_api_management_product" "test" {
+  product_id            = "test-product"
+  api_management_name   = azurerm_api_management.test.name
+  resource_group_name   = azurerm_resource_group.test.name
+  display_name          = "Test Product"
+  subscription_required = false
+  published             = false
+}
+
+resource "azurerm_api_management_tag" "test" {
+  api_management_id = azurerm_api_management.test.id
+  name              = "acctestTag-%d"
+}
+
+resource "azurerm_api_management_product_tag" "test" {
+  product_name        = azurerm_api_management_product.test.product_id
```

Comment:
this is called product_name but take product id?

Reply:
Yes, API doc named it `produceId`, to distinguish with the resource ID like `/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.ApiManagement/service/service1/products/product1/tags/tagId1`,  I think `product_name` is better here.